### DISCUSSION TITLE
Downgrading Firmware

### DISCUSSION THREAD

**Discussion by Vexorify** [roleoroleo/yi-hack-Allwinner-v2#387] [Q&A] [answered]
So I have five of these cameras: 2x QFUS (9.0.36.03_202109231441), 1x 9FUS (8.2.0.0A_202012041528), and 2x IFUS(11.0.36.7_202202170022) models. I recently updated the IFUS models through the official app and motion detection stopped working. The QFUS models have the firmware that worked prior on the IFUS models. Is there anyway I can dump the firmware or download the stock firmware and downgrade?

**Comment by roleoroleo** [comment]
You could use a backup (of the home partition) to restore the previous fw.
But you can't mix firmware from different models.
The stock fw is not available as a download, you can download only the upgrade file but it's not complete.

**Reply by Vexorify** [reply]
Would that not have required me to back it up prior to updating the FW? If not, how would I go about restoring it?

**Reply by roleoroleo** [reply]
I can check if I have a backup for your model.

**Comment by dreimer1986** [comment]
I made a backup, but sadly AFTER I made the upgrade. 2x QFUS here... both at 11.0.36.7_202202170022, too.
I made a issue here: https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/388
So we have one way only, finding someone with a backup we can use, right?
Or is support for the newer 11.x firmware planned?

Maybe the installation howto should say that upgrading is not a good idea. Many here say you should upgrade first and that I did. Now we have a lil problem.
Don't get me wrong here. The hack works fine and the cam is already replacing my other china cam which runs into my AdGuard about 55000 times a day. But these other nice features would be nice to have, too...

**Reply by dreimer1986** [reply]
Could we use other y211ga dumps liie these: https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/316 https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/314 or will that cause a mess?

If you need assistance I can hand you a clean backup AFTER the update. (Added to report)
Likely the current hack completely "disables" the motion sensing. No clue how and why, but I am very sure that the YI Home app showed motion detection before hacking.

**Reply by Vexorify** [reply]
Issue is, I updated before I did the hack. But it's crucial I get back the Motion Detection, since it's completely dead with this new update.

**Reply by dreimer1986** [reply]
I managed to get a fitting v9 firmware backup. Question now... How do I keep my serial? Settings should be resetable I guess?

**Reply by Vexorify** [reply]
Did your serial number get swapped when you restored another backup?

**Reply by dreimer1986** [reply]
I did not try the recover method yet as I am still at work.

**Comment by roleoroleo** [comment]
Sorry but I need to clarify some things.

About IFUS/QFUS cams with firmware 11, what's the previous version installed?
AFAIK, when a cam upgrades to a new version, just the minor version changes:
9.0.36.abc ---> 9.0.36.def
Does the cam upgrade from 9 to 11? Did I understand it right?

If you need a backup of the old version, I can check in my archive.
But please, share with me the new version 11.

Serial number is saved in mtdblock6 and mtdblock7.
And normally the upgrade procedure overwrites only mtdblock4 (backup partition).
If you restore mtdblock4 you will not overwrite the serial number.
With a backup of the upgraded cam I can check which partitions are new.

**Reply by dreimer1986** [reply]
Yes it was 9.0.36.02_202105081647 (close to that at least. I stole this one off mtdblock3.bin ./homever with 7zip) before and now it's 11.0.36.7_202202170022. The guy I got the backup from recommended these cams to me and we found this hack. He got his cams before me and updated + hacked two of his four cam pack. I got a two cam pack of the same offer and updated + hacked both. I can confirm that the datecode off mine was somewhere the same date.

I hope the backups I threw into the report are what you are looking for.

**Reply by roleoroleo** [reply]
> hope the backups I threw into the report are what you are looking for.

Yes, thank you.

After taking a look at the new version, I think that the hack should work properly.
And I think that the motion detection problem is not related to the hack.
To check it, you can simply change the sd card with a clean one and check if motion detection works with the original app.
The difference between the hacked cam (without sd card) and the original cam is this useless piece of code in /backup/init.sh:
```
if [ -f /home/app/lower_half_init.sh ];then
    source /home/app/lower_half_init.sh
else
    source /backup/lower_half_init.sh
fi
```
```
if [ -f /tmp/sd/lower_half_init.sh ];then
    source /tmp/sd/lower_half_init.sh
elif [ -f /home/app/lower_half_init.sh ];then
    source /home/app/lower_half_init.sh
else
    source /backup/lower_half_init.sh
fi
```

**Reply by dreimer1986** [reply]
So all the binaries and stuff are still the same? Crazy... OK, will try that later @ home. If that fails I would try the unbrick way to downgrade? This only changes back mtdblock4.bin right? Or does this contain anything? mtdblock3.bin is different, too though and maybe some other partition (aside the config ones) too?

**Reply by roleoroleo** [reply]
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/388#issuecomment-1063015933

**Reply by dreimer1986** [reply]
So... I removed both SD cards, checked that Motion detection is still on and waved... Nothing. Then I switched it off (Home mode) and on again (Not at home mode) in Original app, works sort of.
![Screenshot_20220309-200532](https://user-images.githubusercontent.com/29841368/157513650-4481195c-3194-47b6-90fa-a742c9f3adcf.jpg)
But I even managed to get this after the cam saw my beautiful face :P Really worth an alarm...
![Screenshot_20220309-201319](https://user-images.githubusercontent.com/29841368/157515050-3ee2ae7b-6b82-4559-bb3d-9aae0b7bccc6.jpg)

Tried the same with card inserted. First nothing, then disable and enable: Nothing... Then I retried the same without SD... works again. So...  looks like the problem is indeed somehow coming from the Hack?

**Reply by dreimer1986** [reply]
Now I just wonder if I switch back or wait a bit to be able to test stuff. But even if I would wnat to go back... No clue how ^^

**Reply by roleoroleo** [reply]
I will check it.

**Comment by dreimer1986** [comment, marked as answer]
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/388#issuecomment-1072224129 check here, revert worked fine that way!

**Reply by Vexorify** [reply]
Hi, sorry for the late response I was really busy. So I posted on the forums about the issue, they didn't suggest any solution on how to fix the issue, so I'm back here trying to fix it. I'm trying to follow your #388 thread, but I'm a bit confused on what files you used to downgrade the IFUS camera back to version 9 to restore the motion detection function.

**Reply by Vexorify** [reply]
Nevermind, figured it out after viewing your thread and #305 .

**Comment by peterek1983** [comment]
Please someone explain step by step how to downgrade IFUS firmware 11.0.36.7 to previous 9.0.36.  
I have clean unhacked camera. 
How to do this and what files put on SD card to downgrade to orginal stock 9.0.36 firmware.
